lib/Optimizer/Transforms/GenDeviceCodeLoader.cpp

```diff
@@ -84,34 +113,27 @@ class GenerateDeviceCodeLoader
         opf.enableDebugInfo(/*enable=*/true,
                             /*pretty=*/false);
         strOut << "module { ";
-        funcOp.print(strOut, opf);
-        strOut << '\n';
 
         // We'll also need any non-inlined functions that are
         // called by our cudaq kernel
         // Set of dependent kernels that we've included.
-        DenseSet<llvm::StringRef> includedNeededFuncs;
-        const std::function<WalkResult(func::CallOp)> emitDependentFunc =
-            [&](func::CallOp callOp) {
-              if (auto neededFunc =
-                      module.lookupSymbol<func::FuncOp>(callOp.getCallee())) {
-                if (neededFunc.empty())
-                  return WalkResult::skip();
-                // If we've included this kernel, e.g. it's called a second
-                // time, skip.
-                if (includedNeededFuncs.contains(neededFunc.getName()))
-                  return WalkResult::skip();
-                neededFunc.print(strOut, opf);
-                includedNeededFuncs.insert(neededFunc.getName());
-                strOut << '\n';
-                // Recursively look into the dependent kernel.
-                neededFunc.walk(emitDependentFunc);
-                return WalkResult::advance();
-              }
-              return WalkResult::advance();
-            };
-
-        funcOp.walk(emitDependentFunc);
+        // Note: the `CallGraphNode` does include 'this' function.
+        mlir::CallGraphNode *node =
+            callGraph.lookupNode(funcOp.getCallableRegion());
+        // Iterate over all dependent kernels starting at this node.
+        for (auto Itr = llvm::df_begin(node), ItrE = llvm::df_end(node);
```

Comment:
nit: don't use CamelCase, use camelCase for variable names.

Reply:
Fixed.